web/client/components/widgets/builder/wizard/map/enhancers/handleNodePropertyChanges.js

```diff
@@ -49,10 +49,25 @@ module.exports = withHandlers({
     changeGroupProperty:
         ({ onChange = () => { }, map = [] }) =>
             (id, key, value) => {
+
+                const EXPANDED = 'expanded';
                 const index = findIndex(map.groups || [], {
                     id
                 });
-                onChange(`map.groups[${index}].${key}`, value);
+
+                if (key === EXPANDED) {
+                    if (!Array.isArray(map.groups)) {
+                        onChange(`map.groups`, []);
+                        onChange(`map.groups[${index === -1 ? 0 : index}].${key}`, value);
+                        onChange(`map.groups[${index === -1 ? 0 : index}].id`, id);
+                    } else {
+                        (!map.groups[index === -1 ? map.groups.length : index]
```

Comment:
Here i have a few questions

why map.groups.length and not map.groups.length - 1

Also when index is -1 ?

Reply:
> Here i have a few questions
> 
> why map.groups.length and not map.groups.length - 1
> Because you get an item of groups by array index 
> Also when index is -1 ?
> When findIndex didn't find an item in array